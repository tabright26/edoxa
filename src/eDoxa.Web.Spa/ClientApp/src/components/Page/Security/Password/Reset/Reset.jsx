import React from "react";
import { Card, CardBody, Col, Container, Row } from "reactstrap";
import { Redirect } from "react-router-dom";
import SecurityPassword from "../../../../../forms/Security/Password";
import withUserContainer from "../../../../../containers/App/User/withUserContainer";
import queryString from "query-string";

const ResetPassword = ({ location, actions }) => {
  const code = queryString.parse(location.search).code;
  if (!code) {
    return <Redirect to="/errors/404" />;
  }
  return (
    <div className="app flex-row align-items-center">
      <Container>
        <Row className="justify-content-center">
          <Col md="9" lg="7" xl="6">
            <Card className="mx-4">
              <CardBody className="p-4">
                <h1>Reset Password</h1>
                <p className="text-muted">Reset your account password</p>
                <SecurityPassword.Reset onSubmit={fields => actions.resetPassword(fields, code)} />
              </CardBody>
            </Card>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default withUserContainer(ResetPassword);
