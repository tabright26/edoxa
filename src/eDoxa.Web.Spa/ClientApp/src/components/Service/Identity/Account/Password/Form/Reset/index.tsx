import React, { FunctionComponent } from "react";
import {
  FormGroup,
  Form,
  InputGroup,
  InputGroupAddon,
  InputGroupText
} from "reactstrap";
import { Field, reduxForm, InjectedFormProps, FormErrors } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { RESET_USER_PASSWORD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { throwSubmissionError } from "utils/form/types";
import { resetUserPassword } from "store/actions/identity";
import { PASSWORD_REQUIRED } from "utils/form/validators";
import { AxiosActionCreatorMeta } from "utils/axios/types";
import { MapStateToProps, connect } from "react-redux";
import { RootState } from "store/types";
import queryString, { ParseOptions } from "query-string";
import { withRouter, RouteComponentProps } from "react-router-dom";
import { push } from "connected-react-router";
import { UserId } from "types/identity";

type FormData = {
  code: string;
  userId: UserId;
  password: string;
  newPassword: string;
};

type Params = {
  userId: UserId;
};

type OwnProps = RouteComponentProps<Params>;

type StateProps = {};

type OutterProps = {};

type InnerProps = InjectedFormProps<FormData, Props> & StateProps;

type Props = InnerProps & OutterProps;

const Reset: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  submitting,
  anyTouched
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary anyTouched={anyTouched} error={error} />
    <Field type="hidden" name="code" component={Input.Text} />
    <Field type="hidden" name="userId" component={Input.Text} />
    <InputGroup size="sm" className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="password"
        placeholder="New password"
        autoComplete="password"
        component={Input.Password}
      />
    </InputGroup>
    <InputGroup size="sm" className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="newPassword"
        placeholder="Confirm new password"
        autoComplete="new-password"
        component={Input.Password}
      />
    </InputGroup>
    <FormGroup className="mb-0">
      <Button.Submit loading={submitting} block>
        Reset
      </Button.Submit>
    </FormGroup>
  </Form>
);

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (
  _state,
  ownProps
) => {
  const options: ParseOptions = {
    decode: false
  };
  const { code } = queryString.parse(ownProps.location.search, options);
  return {
    initialValues: { code, userId: ownProps.match.params.userId }
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withRouter,
  connect(mapStateToProps),
  reduxForm<FormData, Props>({
    form: RESET_USER_PASSWORD_FORM,
    onSubmit: async (values, dispatch) => {
      try {
        return await new Promise((resolve, reject) => {
          const meta: AxiosActionCreatorMeta = { resolve, reject };
          dispatch(resetUserPassword(values, meta));
        });
      } catch (error) {
        throwSubmissionError(error);
      }
    },
    onSubmitSuccess: (_result, dispatch) => {
      dispatch(push("/authentication/login"));
    },
    validate: values => {
      const errors: FormErrors<FormData> = {};
      if (!values.password) {
        errors.password = PASSWORD_REQUIRED;
      } else if (values.password !== values.newPassword) {
        errors._error = "Confirm password doesn't match password";
      }
      return errors;
    }
  })
);

export default enhance(Reset);
