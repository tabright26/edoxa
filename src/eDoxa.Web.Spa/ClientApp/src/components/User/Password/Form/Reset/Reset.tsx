import React, { FunctionComponent } from "react";
import {
  FormGroup,
  Form,
  InputGroup,
  InputGroupAddon,
  InputGroupText
} from "reactstrap";
import { Field, reduxForm, InjectedFormProps } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { RESET_USER_PASSWORD_FORM } from "forms";
import { compose } from "recompose";
import FormValidation from "components/Shared/Form/Validation";
import { throwSubmissionError } from "utils/form/types";
import { resetUserPassword } from "store/actions/identity/actions";
import {
  emailRegex,
  passwordRegex,
  EMAIL_REQUIRED,
  EMAIL_INVALID,
  PASSWORD_REQUIRED,
  PASSWORD_INVALID
} from "validation";
import { AxiosActionCreatorMeta } from "utils/axios/types";

interface Props {}

interface FormData {}

const validate = values => {
  const errors: any = {};
  if (!values.email) {
    errors.email = EMAIL_REQUIRED;
  } else if (!emailRegex.test(values.email)) {
    errors.email = EMAIL_INVALID;
  }
  if (!values.password) {
    errors.password = PASSWORD_REQUIRED;
  } else if (!passwordRegex.test(values.password)) {
    errors.password = PASSWORD_INVALID;
  }
  return errors;
};

async function submit(values, dispatch) {
  try {
    return await new Promise((resolve, reject) => {
      const meta: AxiosActionCreatorMeta = { resolve, reject };
      dispatch(resetUserPassword(values, meta));
    });
  } catch (error) {
    throwSubmissionError(error);
  }
}

const ResetUserPasswordForm: FunctionComponent<InjectedFormProps<FormData> &
  Props &
  any> = ({ handleSubmit, handleCancel, dispatch, error }) => (
  <Form
    onSubmit={handleSubmit(data =>
      submit(data, dispatch).then(() => handleCancel())
    )}
  >
    {error && <FormValidation error={error} />}
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>@</InputGroupText>
      </InputGroupAddon>
      <Field type="text" name="email" label="Email" component={Input.Text} />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="password"
        label="Password"
        component={Input.Password}
      />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="confirmPassword"
        label="Confirm Password"
        component={Input.Password}
      />
    </InputGroup>
    <FormGroup className="mb-0">
      <Button.Submit block>Reset</Button.Submit>
    </FormGroup>
  </Form>
);

const enhance = compose<any, any>(
  reduxForm<FormData, Props>({
    form: RESET_USER_PASSWORD_FORM,
    validate
  })
);

export default enhance(ResetUserPasswordForm);
