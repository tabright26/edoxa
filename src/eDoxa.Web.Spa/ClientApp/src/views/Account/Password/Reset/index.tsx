import React, { FunctionComponent } from "react";
import { Card, CardBody } from "reactstrap";
import PasswordForm from "components/Service/Identity/Password/Form";

const Reset: FunctionComponent = () => (
  <Card className="mx-4">
    <CardBody className="p-4">
      <h1 className="mb-3">Reset password</h1>
      <PasswordForm.Reset />
    </CardBody>
  </Card>
);

export default Reset;
