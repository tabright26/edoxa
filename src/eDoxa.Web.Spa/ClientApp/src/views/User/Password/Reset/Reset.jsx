import React from "react";
import { Card, CardBody } from "reactstrap";
import { Redirect } from "react-router-dom";
import SecurityPassword from "../../../../forms/User/Password";
import withUserContainer from "../../../../containers/withUserContainer";
import queryString from "query-string";

const ResetPassword = ({ location, actions }) => {
  const code = queryString.parse(location.search).code;
  if (!code) {
    return <Redirect to="/errors/404" />;
  }
  return (
    <Card className="mx-4">
      <CardBody className="p-4">
        <h1>Reset Password</h1>
        <p className="text-muted">Reset your account password</p>
        <SecurityPassword.Reset onSubmit={fields => actions.resetPassword(fields, code)} />
      </CardBody>
    </Card>
  );
};

export default withUserContainer(ResetPassword);
