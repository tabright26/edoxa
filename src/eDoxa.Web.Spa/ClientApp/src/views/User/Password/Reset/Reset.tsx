import React, { FunctionComponent } from "react";
import { Redirect } from "react-router-dom";
import { Card, CardBody } from "reactstrap";
import queryString from "query-string";
import { withtUser } from "store/root/user/container";
import PasswordForm from "forms/User/Password";
import { compose } from "recompose";

const ResetPassword: FunctionComponent<any> = ({ location, actions }) => {
  const code = queryString.parse(location.search).code;
  if (!code) {
    return <Redirect to="/errors/404" />;
  }
  return (
    <Card className="mx-4">
      <CardBody className="p-4">
        <h1>Reset Password</h1>
        <p className="text-muted">Reset your account password</p>
        <PasswordForm.Reset onSubmit={fields => actions.resetPassword(fields, code)} />
      </CardBody>
    </Card>
  );
};

const enhance = compose<any, any>(withtUser);

export default enhance(ResetPassword);
