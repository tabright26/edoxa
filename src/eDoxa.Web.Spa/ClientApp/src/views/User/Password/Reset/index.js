import ResetPassword from "./Reset";

export default ResetPassword;
