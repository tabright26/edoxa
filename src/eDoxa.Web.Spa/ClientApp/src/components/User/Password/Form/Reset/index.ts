import Form from "./Reset";

export default Form;
