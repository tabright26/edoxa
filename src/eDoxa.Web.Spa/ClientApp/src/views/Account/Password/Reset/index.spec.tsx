import React from "react";
import renderer from "react-test-renderer";
import Reset from ".";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "store";

it("renders correctly", () => {
  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Reset />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
