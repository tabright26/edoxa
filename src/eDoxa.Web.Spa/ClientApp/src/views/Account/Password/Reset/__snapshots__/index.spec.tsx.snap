// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mx-4 card"
>
  <div
    className="p-4 card-body"
  >
    <h1
      className="mb-3"
    >
      Reset password
    </h1>
    <form
      className=""
      onSubmit={[Function]}
    >
      <input
        className="form-control-sm form-control"
        name="code"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={null}
        type="hidden"
        value=""
      />
      <div
        className="invalid-feedback"
      />
      <input
        className="form-control-sm form-control"
        name="userId"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={null}
        type="hidden"
        value=""
      />
      <div
        className="invalid-feedback"
      />
      <div
        className="mb-3 input-group input-group-sm"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="icon-lock"
            />
          </span>
        </div>
        <input
          autoComplete="password"
          className="form-control-sm form-control"
          disabled={false}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="New password"
          type="password"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Password is required
        </div>
      </div>
      <div
        className="mb-3 input-group input-group-sm"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            <i
              className="icon-lock"
            />
          </span>
        </div>
        <input
          autoComplete="new-password"
          className="form-control-sm form-control"
          disabled={false}
          name="newPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Confirm new password"
          type="password"
          value=""
        />
        <div
          className="invalid-feedback"
        />
      </div>
      <div
        className="mb-0 form-group"
      >
        <button
          aria-label={null}
          className="btn btn-primary btn-block"
          disabled={null}
          onClick={[Function]}
          type="submit"
        >
          Reset
        </button>
      </div>
    </form>
  </div>
</div>
`;
